node/commons/compatibility-layer.ts

```diff
@@ -24,8 +26,8 @@ export const convertBiggyProduct = (
 
   return {
     categories,
-    cacheId: product.link,
-    productId: product.product || product.id,
+    cacheId: isProductSplit ? `${product.link}-${product.id}` : product.link,
+    productId: product.id,
```

Comment:
This needs to be `product.product`, right? `product.id` is our elastic search ID, that in split cases is not equal to catalog's `productId`.

```suggestion
    productId: product.product,
```

Reply:
It is not that simple.
The `productId` field can not be repeated. It is used as `key` in some react lists. When the store has the split enabled it will be a problem.